src/noises.jl

```diff
@@ -24,19 +24,147 @@ struct NetworkPauliNoise
+                # multiple qubits are being acted on
+                # find the qubit with the maximum time accounted for, the other qubits must be 'brought up to speed' with it
+                max_time = maximum([accounted_for_time[q] for q in 
+                acted_on_qubits]) # get the maximum time accounted for
+
+                # for every qubit that is not up to speed with the max time
+                for q in acted_on_qubits
+                    if accounted_for_time[q] < max_time
+                        # this qubit has not been accounted for up until now, so we must add T1 and T2 noise here now
+                        
+                        # get the error operations from time idling, t1, t2 values, and push them
+                        idle_time = max_time - accounted_for_time[q]
+                        λ₁, λ₂ = thermal_relaxation_error_rate(error.t1, error.t2, idle_time)
+                        push!(noisy_circuit, T1NoiseOp(q,λ₁), T2NoiseOp(q,λ₂))
+
+                        # now update its time
+                        accounted_for_time[q] = max_time # set it to the maximum time, so that it is now in sync with the other qubits
+                    end
+                end
+            end
+
+            # now all important qubits are up to speed, we can apply the operation 
+            push!(noisy_circuit, op)
+
+            # and update the time elapsed for this operation 
+            for q in acted_on_qubits
+                accounted_for_time[q] += time_to_elapse # update the time for all qubits that were acted on
+            end
+
+            ## --- After applying operation ---
+        end
+    end
+
+    # Now all operations have T1T2 noise padded before, and in-between them if needed. So now we need to bring all of the qubits up to speed with the end time.
+    # Get the max time on a qubit (this is the time that it takes to execute the entire circuit)
+    max_time = maximum(accounted_for_time)
+
+    # Bring all qubits up to speed with the maximum time (pad with T1T2)
+    for q in qubit_list
+        if accounted_for_time[q] < max_time  
+            # this qubit has not been accounted for up until now, so we must add T1 and T2 noise here now
+            
+            # get the error operations from time idling, t1, t2 values, and push them
+            idle_time = max_time - accounted_for_time[q]
+            λ₁, λ₂ = thermal_relaxation_error_rate(error.t1, error.t2, idle_time)
+            push!(noisy_circuit, T1NoiseOp(q,λ₁), T2NoiseOp(q,λ₂))
+        end
+    end
+
+    # TODO: make sure we can simplify/compactify_circuit with QuantumClifford
+
+    return noisy_circuit
+end
+    
+# thermal relaxation error from t1 and t2 values, and the time elapsed. From YipiaoWu/QuantumHardware
+function thermal_relaxation_error_rate(t1, t2, gate_time) 
+    λ₁ = 1 - exp(-gate_time/t1)
+    t_ϕ = t1*t2 / (2*t1 - t2)
+    λ₂ = 1 - exp(-gate_time/t_ϕ)
+    return λ₁, λ₂
+end
```

Comment:
`T1(x) * T1(x) = T1(2x)` is a constraint that needs to be obeyed. T1 is a superoperator, but how does this equality translate for the lambda parameter.

If lambda is the probability of an event then `lambda_2x = 2*lambda_x*(1-lambda_x)` -- if I am understanding what these rate are, the equation must be obeyed, so lets try it as a test, and we will revisit if that test is failing.